client/src/pages/500.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import SunBear from '../shared/icons/SunBear';
+import Button from '../shared/components/Button';
+import { useRouter } from 'next/router';
+
+const ServerCrash = (): JSX.Element => {
+  const router = useRouter();
+
+  return (
+    <div className="bg-[#FDFBFB] h-screen flex justify-center items-center">
+      <div
+        className="py-32 px-20 rounded-[5px] w-[41rem] flex justify-center"
+        style={{
+          background:
+            'linear-gradient(180deg, rgba(245, 211, 207, 0) 0%, rgba(245, 211, 207, 0.34) 59%, #F5D3CF 85%)',
+        }}
+      >
+        <div>
+          <div className="flex gap-4">
+            <SunBear height={120} width={120} />
+            <div className="flex flex-col gap-2 items-start">
+              <div className="text-primary-base text-[4rem] font-semibold leading-[4.96rem]">
+                500
+              </div>
+              <div className="flex gap-1 flex-col text-dark">
+                <div className="text-xl font-semibold leading-[1.55rem]">
+                  Oops, something went wrong.
+                </div>
+                <div className="text-base leading-[1.5rem]">
+                  The server crashed. Try to refresh this page.
+                </div>
+              </div>
+              <Button
+                text="Go back to home page"
+                buttonClass="text-primary-base text-xs leading-[1.125rem] font-medium border border-primary-base rounded-[5px] py-1 px-4"
+                onClick={() => {
+                  void router.replace('/');
+                }}
+              />
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
```

Comment:
I see you are using the same layout in 404 as well, to make it more maintainable,  instead of repeating it, please create a separate component for the layout, and probably pass the difference as children to the component

something like :
```
  <ErrorPageLayout>
       <div className="flex flex-col gap-2 items-start">
              <div className="text-primary-base text-[4rem] font-semibold leading-[4.96rem]">
                500
              </div>
              <div className="flex gap-1 flex-col text-dark">
                <div className="text-xl font-semibold leading-[1.55rem]">
                  Oops, something went wrong.
                </div>
                <div className="text-base leading-[1.5rem]">
                  The server crashed. Try to refresh this page.
                </div>
              </div>
  </ErrorPageLayout>
```